pkg/processor/build/builder.go

```diff
@@ -116,14 +116,18 @@ type Builder struct {
 	originalFunctionConfig functionconfig.Config
 
 	s3Client common.S3Client
+
+	// default timeout before bailing build process
+	defaultBuildTimeoutSeconds int64
 }
 
 // NewBuilder returns a new builder
 func NewBuilder(parentLogger logger.Logger, platform platform.Platform, s3Client common.S3Client) (*Builder, error) {
 	newBuilder := &Builder{
-		logger:   parentLogger,
-		platform: platform,
-		s3Client: s3Client,
+		logger:                     parentLogger,
+		platform:                   platform,
+		s3Client:                   s3Client,
+		defaultBuildTimeoutSeconds: 3600,
```

Comment:
Move 3600 inside `getBuildTimeoutSeconds()`
This doesn't enable any flow where this is somehow more flexible or configurable - so let's keep this contained